src/OrchardCore.Modules/Orchard.Tokens/StandardTokens.cs

```diff
@@ -8,13 +9,13 @@ namespace Orchard.Tokens
 {
     public static class StandardTokens
     {
-        public static void RegisterStandardTokens(this IHandlebars handlebars)
+        public static void RegisterStandardTokens(this IHandlebars handlebars, IHttpContextAccessor httpContextAccessor)
         {
             handlebars.RegisterHelper("dateformat", (output, context, arguments) =>
             {
-                IServiceProvider serviceProvider = context.ServiceProvider;
-                var clock = serviceProvider.GetRequiredService<IClock>();
-                var siteService = serviceProvider.GetRequiredService<ISiteService>();
+                var services = httpContextAccessor.HttpContext.RequestServices;
```

Comment:
Tokens are not resolved by DI, this is exactly why we initialize them with the IHttpContextAccessor.